tools/ack-discover/ecrpublic.py

```diff
@@ -0,0 +1,136 @@
+    secret_access_key: str
+    session_token: str
+
+
+def _assume_reader_role(writer):
+    """Returns an access key ID, secret access key, and a security token after
+    assuming an ECR Public readonly role.
+    """
+    writer.debug("[ecr.assume_reader_role] assuming ECR Public ReadOnly role ...")
+    role_session_name = str(uuid.uuid4())
+    sts_client = boto3.client("sts")
+    resp = sts_client.assume_role(
+        RoleArn=ECR_PUBLIC_READER_ROLE_ARN,
+        RoleSessionName=role_session_name,
+        DurationSeconds=60*60,  # 1 hour...
+    )
+    creds = resp['Credentials']
+    return AWSCredentials(
+        access_key_id=creds['AccessKeyId'],
+        secret_access_key=creds['SecretAccessKey'],
+        session_token=creds['SessionToken'],
+    )
+
+
+def get_client(writer):
+    """Returns the ECR Public client object after assuming the ECR Public
+    Reader role.
+    """
+    aws_creds = _assume_reader_role(writer)
+
+    # NOTE(jaypipes): ECR Public requires "us-east-1" when calling
+    # DescribeRepositories...
+    return boto3.client(
+        "ecr-public",
+        region_name="us-east-1",
+        aws_access_key_id=aws_creds.access_key_id,
+        aws_secret_access_key=aws_creds.secret_access_key,
+        aws_session_token=aws_creds.session_token,
+    )
+
+
+@dataclasses.dataclass
+class Repository:
+    registry_id: str
+    name: str
+    uri: str
+    created_on: str
+
+
+def get_repository(writer, ep_client, repo_url):
+    """Returns an Repository describing the supplied repository
+    URL or None if no such repository exists.
+    """
+    writer.debug("[get_public_repository] ", repo_url)
+    parts = repo_url.split("/")
+    registry_id = ECR_PUBLIC_AWS_ACCOUNT_ID
+    repo_name = parts[2]
+    try:
+        repos = ep_client.describe_repositories(
+            registryId=registry_id,
+            repositoryNames=[repo_name],
+        )
+        repo_data = repos["repositories"][0]
+        return Repository(
+            registry_id=registry_id,
+            name=repo_name,
+            uri=repo_data["repositoryUri"],
+            created_on=repo_data["createdAt"],
+        )
+    except ep_client.exceptions.RepositoryNotFoundException:
+        return None
+
+
+def get_repository_latest_tag(ep_client, repo):
+    """Returns the image tag of the latest pushed image in a supplied repo."""
+    most_recent = None
+    latest_tag = None
+    try:
+        images = ep_client.describe_images(
+            registryId=repo.registry_id,
+            repositoryName=repo.name,
+        )
+        for image in images["imageDetails"]:
+            pushed_at = image["imagePushedAt"]
+            if most_recent is None or pushed_at > most_recent:
+                most_recent = pushed_at
+                latest_tag = image["imageTags"][0]
+        return latest_tag
+    except ep_client.exceptions.RepositoryNotFoundException:
+        return latest_tag
+
+
+def chart_has_stable_tag(ep_client, chart_repo):
```

Comment:
this can be replaced by just checking for a "stable" branch on github repo too. right?

Reply:
I suppose so, yes...